python_modules/dagster/dagster/_core/definitions/data_version.py

```diff
@@ -403,16 +403,19 @@ def get_current_data_version(
 
     @cached_method
     def _get_status(self, key: "AssetKeyPartitionKey") -> StaleStatus:
-        current_version = self._get_current_data_version(key=key)
-        if current_version == NULL_DATA_VERSION:
-            return StaleStatus.MISSING
-        elif self.asset_graph.is_partitioned(key.asset_key) and not key.partition_key:
-            return StaleStatus.FRESH
-        elif self.asset_graph.is_source(key.asset_key):
+        # The status loader does not support querying for the stale status of a
+        # partitioned asset without specifying a partition, so we return here.
+        if self.asset_graph.is_partitioned(key.asset_key) and not key.partition_key:
```

Comment:
just to confirm - the NULL_DATA_VERSION case is never relevant for these assets?

Reply:
I don't believe so-- `NULL_DATA_VERSION` is relevant for individual partitions, but the status of a partitioned asset with no specified partition key is undefined, so really we can return whatever dummy value we want. If frontend is querying for this it should be ignoring the return value.